joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/utils/OffsetUtils.scala

```diff
@@ -0,0 +1,30 @@
+package io.joern.x2cpg.utils
+
+import io.joern.x2cpg.AstNodeBuilder.ZeroIndexedCoordinates
+
+object OffsetUtils {
+  def getLineOffsetIndex(fileContent: Option[String]): Array[Int] = {
+    fileContent
+      .map { content =>
+        var totalCharsCounted = 0
+        content.linesWithSeparators.map { line =>
+          val lineStartOffset = totalCharsCounted
+          totalCharsCounted += line.length()
+          lineStartOffset
+        }.toArray
+      }
+      .getOrElse(Array.empty)
+  }
+
+  def coordinatesToOffset(lineOffsetIndex: Array[Int], coordinates: ZeroIndexedCoordinates): Option[(Int, Int)] = {
+    coordinates.match {
+      case ZeroIndexedCoordinates(startLine, startColumn, endLine, endColumn)
+          if endLine <= lineOffsetIndex.length - 1 =>
```

Comment:
The length check will just hide bugs in other parts of the code. If for some reason the line numbers of certain frontends are out of range, the frontend should deal with it. But most likely it will just be a bug.

Reply:
Return type is than just `(Int, Int)`